src/cryptsetup/cryptsetup.c

```diff
@@ -1503,10 +1503,7 @@ static int run(int argc, char *argv[]) {
 
                 flags = determine_flags();
 
-                if (arg_timeout == USEC_INFINITY)
-                        until = 0;
-                else
-                        until = now(CLOCK_MONOTONIC) + arg_timeout;
+                until = usec_add(now(CLOCK_MONOTONIC), arg_timeout);
```

Comment:
The ask_password_xyz() functions currently take an until param of 0 as meaning "wait forever". hence this changes behaviour quite a bit. That said, those functions should probably be fixed to make "0" mean, don#t wait at all, and USEC_INFINITY mean wait forever